processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -1309,4 +1269,136 @@ private void writeMetadataToFile(File metadataFile, Map<String, Object> metadata
     }
     IndexIO.checkFileSize(metadataFile);
   }
+
+  static class MergeIterator implements Iterator<String>
```

Comment:
can we give this a more specific name as I think there is already a merge iterator in Druid